plugins/inventory/digitalocean.py

```diff
@@ -0,0 +1,215 @@
+author:
+  - Janos Gerzson (@grzs)
+  - Tadej Borovšak (@tadeboro)
+short_description: DigitalOcean Inventory Plugin
+version_added: "1.1.0"
+description:
+  - DigitalOcean (DO) inventory plugin.
+  - Acquires droplet list from DO API.
+  - Uses configuration file that ends with '(do_hosts|digitalocean|digital_ocean).(yaml|yml)'.
+extends_documentation_fragment:
+  - community.digitalocean.digital_ocean.documentation
+  - constructed
+  - inventory_cache
+options:
+  plugin:
+    description:
+      - The name of the DigitalOcean Inventory Plugin,
+        this should always be C(community.digitalocean.digitalocean).
+    required: true
+    choices: ['community.digitalocean.digitalocean']
+  api_token:
+    description:
+     - DigitalOcean OAuth token.
+    required: true
+    type: str
+    aliases: [ oauth_token ]
+    env:
+      - name: DO_API_TOKEN
+  attributes:
+    description: >-
+      Droplet attributes to add as host vars to each inventory host.
+      Check out the DO API docs for full list of attributes at
+      U(https://developers.digitalocean.com/documentation/v2/#list-all-droplets).
+    type: list
+    elements: str
+    default:
+      - id
+      - name
+      - networks
+      - region
+      - size_slug
+  var_prefix:
+    description:
+      - Prefix of generated varible names (e.g. C(tags) -> C(do_tags))
+      - Has to start with a letter,
+        the only valid special character is underscore (C(_)).
+    type: str
+    default: 'do_'
+  pagination:
+    description:
+      - Maximum droplet objects per response page.
+      - If the number of droplets related to the account exceeds this value,
+        the query will be broken to multiple requests (pages).
+      - This value cannot be greater than 200.
+    type: int
+    default: 200
+'''
+
+EXAMPLES = r'''
+# Using keyed groups and compose for hostvars
+plugin: community.digitalocean.digitalocean
+api_token: "{{ api_token }}"
+attributes:
+  - id
+  - name
+  - memory
+  - vcpus
+  - disk
+  - size
+  - image
+  - networks
+  - volume_ids
+  - tags
+  - region
+keyed_groups:
+  - key: do_region.slug
+    prefix: 'region'
+    separator: '_'
+  - key: do_tags | lower
+    prefix: ''
+    separator: ''
+compose:
+  ansible_host: do_networks.v4 | selectattr('type','eq','public')
+    | map(attribute='ip_address') | first
+  class: do_size.description | lower
+  distro: do_image.distribution | lower
+'''
+
+import re
+import json
+from ansible.errors import AnsibleParserError
+from ansible.module_utils.urls import Request
+from ansible.module_utils.six.moves.urllib.error import URLError, HTTPError
+from ansible.plugins.inventory import BaseInventoryPlugin, Constructable, Cacheable
+
+
+class InventoryModule(BaseInventoryPlugin, Constructable, Cacheable):
+
+    NAME = 'community.digitalocean.digitalocean'
+
+    def _validate_config(self, config):
+        if 'var_prefix' in config:
+            if not re.match('^[a-z][a-z_]*$', config['var_prefix']):
```

Comment:
I'd probably leave this away. If you really want to kee pit, you should also allow digits after the first letter.

Reply:
I deleted the _verify_config block. So no more questions left I think.